libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/Tx/Body.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Api.Tx.Body (spec) where
+
+import Cardano.Ledger.Api.Era
+import Cardano.Ledger.Api.Tx.Body
+import Cardano.Ledger.Coin
+import Cardano.Ledger.Compactible
+import Cardano.Ledger.DPState
+import Cardano.Ledger.PoolParams
+import Cardano.Ledger.Shelley.Core
+import Cardano.Ledger.Shelley.Delegation.Certificates
+import Cardano.Ledger.Shelley.UTxO hiding (consumed, produced)
+import qualified Cardano.Ledger.UMapCompact as UM
+import Cardano.Ledger.Val
+import Data.Foldable
+import qualified Data.Map.Strict as Map
+import qualified Data.Sequence.Strict as SSeq
+import qualified Data.Set as Set
+import Lens.Micro
+import Test.Cardano.Ledger.Babbage.Arbitrary ()
+import Test.Cardano.Ledger.Common
+
+totalTxDeposits ::
+  ShelleyEraTxBody era =>
+  PParams era ->
+  DPState (EraCrypto era) ->
+  TxBody era ->
+  Coin
+totalTxDeposits pp dpstate txb =
```

Comment:
maybe worth leaving a note that this includes both stake credential and stake pool registration deposits? (and DRep in the future...) I can see myself in the future asking "is this _really_ all the deposits"?

Reply:
That's a great point. I'll leave a comment about this. 

This makes me realize that we will have to make `getProducedValue` part of the `EraUTxO` type class, because we will have DRep deposits in Conway.

Reply:
Added a bit of haddock with this info

Reply:
Another question to think about is where will the DRep info reside in the state? Will we need a RState for Reps, and the DPState becomes the DPRState?